### DISCUSSION TITLE
Signal Drop behavior

### DISCUSSION THREAD

**Discussion by serzhiio** [DioxusLabs/dioxus#1977] [Q&A]
I have a component which used `use_state` shared state and when this component Dropped `use_state` value was also dropped. Now with `use_signal` it doesn't drop on component drop. 
Is it normal behavior?

UPD: I can use `use_hook` instead, to have required Drop of stored value

**Comment by ealmloff** [comment]
Any signals created within a component with `use_signal` or `Signal::new` should be owned by the component and dropped when the component is dropped

**Comment by serzhiio** [comment]
In my case it doesnt drop with a component.

**Reply by ealmloff** [reply]
Could you open an issue with the code that causes this behavior?
